yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -611,65 +613,70 @@ def pixelize_cylinder(np.float64_t[:,:] buff,
-
-        while theta_i < theta0 + dtheta_i:
-            r_i = r0 - dr_i
-            costheta = math.cos(theta_i)
-            sintheta = math.sin(theta_i)
-            while r_i < r0 + dr_i:
-                if rmax <= r_i:
+    with nogil:
+        for i in range(radius.shape[0]):
+            r0 = radius[i]
+            theta0 = theta[i]
+            dr_i = dradius[i]
+            dtheta_i = dtheta[i]
+            # Skip out early if we're offsides, for zoomed in plots
+            if r0 + dr_i < rbounds[0] or r0 - dr_i > rbounds[1]:
+                continue
+            theta_i = theta0 - dtheta_i
+            theta_inc = r_inc / (r0 + dr_i)
+
+            while theta_i < theta0 + dtheta_i:
+                r_i = r0 - dr_i
+                costheta = math.cos(theta_i)
+                sintheta = math.sin(theta_i)
+                while r_i < r0 + dr_i:
+                    if rmax <= r_i:
+                        r_i += r_inc
+                        continue
+                    y = r_i * costheta
+                    x = r_i * sintheta
+                    pi = <int>((x - x0)/dx)
+                    pj = <int>((y - y0)/dy)
+                    if pi >= 0 and pi < buff.shape[0] and \
+                       pj >= 0 and pj < buff.shape[1]:
+                        # we got a pixel that intersects the grid cell
+                        # now check that this pixel doesn't go beyond the data domain
+                        xp = x0 + pi*dx
+                        yp = y0 + pj*dy
+                        corners[0] = xp*xp + yp*yp
+                        corners[1] = xp*xp + (yp+dy)**2
+                        corners[2] = (xp+dx)**2 + yp*yp
+                        corners[3] = (xp+dx)**2 + (yp+dy)**2
+                        prbounds[0] = prbounds[1] = corners[3]
+                        for i1 in range(3):
+                            prbounds[0] = fmin(prbounds[0], corners[i1])
+                            prbounds[1] = fmax(prbounds[1], corners[i1])
+                        prbounds[0] = math.sqrt(prbounds[0])
+                        prbounds[1] = math.sqrt(prbounds[1])
+
+                        corners[0] = math.atan2(xp, yp)
+                        corners[1] = math.atan2(xp, yp+dy)
+                        corners[2] = math.atan2(xp+dx, yp)
+                        corners[3] = math.atan2(xp+dx, yp+dy)
+                        ptbounds[0] = ptbounds[1] = corners[3]
+                        for i1 in range(3):
+                            ptbounds[0] = fmin(ptbounds[0], corners[i1])
+                            ptbounds[1] = fmax(ptbounds[1], corners[i1])
+
+                        # shift to a [0, PI] interval
+                        ptbounds[0] = ptbounds[0] % twoPI
+                        if ptbounds[0] < 0:
+                            ptbounds[0] += NPY_PI
+                        ptbounds[1] = ptbounds[1] % twoPI
+                        if ptbounds[1] < 0:
+                            ptbounds[1] += NPY_PI
```

Comment:
so I'm not sure why I needed to add the check for negative numbers here, but  after switching from `ptbounds[1] % (2*np.pi)` to `ptbounds[1] % twoPI`, the result returns numbers in the range (-pi,pi).

Reply:
I think we need to understand this bit, and possibly come up with a way to perform the computation that doesn't involve branching; hard-wiring a somewhat expensive calculation generally hurts performance less than ruining branch prediction.
I'm writing this comment early in my review but I'll try to come up with more insight.

Reply:
can you try this patch
```patch
diff --git a/yt/utilities/lib/pixelization_routines.pyx b/yt/utilities/lib/pixelization_routines.pyx
index 194ce7af6..6728375d0 100644
--- a/yt/utilities/lib/pixelization_routines.pyx
+++ b/yt/utilities/lib/pixelization_routines.pyx
@@ -664,12 +664,8 @@ def pixelize_cylinder(np.float64_t[:,:] buff,
                             ptbounds[1] = fmax(ptbounds[1], corners[i1])
 
                         # shift to a [0, PI] interval
-                        ptbounds[0] = ptbounds[0] % twoPI
-                        if ptbounds[0] < 0:
-                            ptbounds[0] += NPY_PI
-                        ptbounds[1] = ptbounds[1] % twoPI
-                        if ptbounds[1] < 0:
-                            ptbounds[1] += NPY_PI
+                        ptbounds[0] = (ptbounds[0]+twoPI) % twoPI
+                        ptbounds[1] = (ptbounds[1]+twoPI) % twoPI
 
                         if prbounds[0] >= rmin and prbounds[1] <= rmax and \
                            ptbounds[0] >= tmin and ptbounds[1] <= tmax:
```
the result should be the same in all cases, but it avoids branching, so I expect it to be faster.
If this works, we'll want to add a comment for why it's done this way.

Reply:
I am wondering if we could use `fmod` here.

Reply:
looks like in cython, `%` is an alias to `fmod` already when using C functions, which returns a negative number when the first argument is negative (which is why the tests started failing here I think).  but I do think it's clearer to use `fmod` to avoid the confusion in the first place (so `fmod(ptbounds[0]+twoPI, twoPI)` to handle the negative numbers). In case it's helpful, I wrote up some scratch code to double check, here's a plot:

![comparison-cython-modulo](https://github.com/user-attachments/assets/811cbaed-f3ce-4d8f-9e26-c095d21ba303)

Reply:
What is this plot showing ? I'm not seeing labels on those axes